src/SignupForm.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+
+class SignupForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      firstName: '',
+      lastName: '',
+      password: '',
+      agreeWithTerms: false,
+    };
+
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
+    this.handleLastNameChange = this.handleLastNameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.toogleAgreeWithTerms = this.toogleAgreeWithTerms.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleEmailChange(e) {
+    this.setState({ email: e.target.value })
+  }
+
+  handleFirstNameChange(e) {
+    this.setState({ firstName: e.target.value })
+  }
+
+  handleLastNameChange(e) {
+    this.setState({ lastName: e.target.value })
+  }
+
+  handlePasswordChange(e) {
+    this.setState({ password: e.target.value })
+  }
+
+  toogleAgreeWithTerms() {
+    this.setState({ agreeWithTerms: !this.state.agreeWithTerms })
```

Comment:
Your new state depends on the previous state, yet you can't be sure it'll be the one you need when using this syntax; Use functional set state instead